src/main/java/games/strategy/engine/data/TripleaHandler.java

```diff
@@ -0,0 +1,475 @@
+          handleDelegate(attributes);
+          break;
+        case XML_NODE_NAME_STEP:
+          handleStep(attributes);
+          break;
+        case "playerProduction":
+          handlePlayerProduction(attributes);
+          break;
+        case XML_NODE_NAME_COST:
+          handleCost(attributes);
+          break;
+        case XML_NODE_NAME_RESULT:
+          handleResult(attributes);
+          break;
+        case XML_NODE_NAME_FRONTIER_RULES:
+          handleFrontierRules(attributes);
+          break;
+        case XML_NODE_NAME_OPTION:
+          handleOption(attributes);
+          break;
+        case XML_NODE_NAME_TERRITORY_OWNER:
+          handleTerritoryOwner(attributes);
+          break;
+        case XML_NODE_NAME_UNIT_PLACEMENT:
+          handleUnitPlacement(attributes);
+          break;
+        case "resourceGiven":
+          handleResourceGiven(attributes);
+          break;
+        case XML_NODE_NAME_PROPERTY:
+          handleProperty(attributes);
+          break;
+        case XML_NODE_NAME_NUMBER:
+          handleNumber();
+          break;
+        case XML_NODE_NAME_BOOLEAN:
+          handleBoolean();
+          break;
+        case "tech":
+          handleTech(attributes);
+          break;
+        case XML_NODE_NAME_PRODUCTION_FRONTIER:
+          handleProductionFrontier(attributes);
+          break;
+        case XML_NODE_NAME_ATTACHMENT:
+          handleAttachment(attributes);
+          break;
+        case XML_NODE_NAME_VALUE:
+        case XML_NODE_NAME_SEQUENCE:
+        case XML_NODE_NAME_RESOURCE_LIST:
+        case XML_NODE_NAME_UNIT_LIST:
+        case "relationshipTypes":
+        case "territoryEffectList":
+        case "technology":
+        case XML_NODE_NAME_OWNER_INITIALIZE:
+        case XML_NODE_NAME_UNIT_INITIALIZE:
+        case "resouceInitialize":
+        case "technologies":
+        case "playerTech":
+        case "category":
+        default:
+          break;
+      }
+    } catch (GameParseException e) {
+      throw new SAXException(e);
+    }
+  }
+
+  private void checkParentNode(String qname) throws GameParseException {
+    if (!stack.isEmpty()) {
+      String actualName = stack.peek().getName();
+      Collection<String> parents = requiredParents.get(qname);
+      if (parents.isEmpty()) {
+        throw new GameParseException("Unknown element '" + qname + "'");
+      } else if (!parents.contains(actualName)) {
+        throw new GameParseException(
+            "Parent of '" + qname + "' must be one of " + Arrays.toString(parents.toArray()) + ", but was '"
+                + actualName + "'");
+      }
+    } else if (!qname.equals(XML_NODE_NAME_GAME)) {
+      throw new GameParseException("Root element must be 'game', found '" + qname + "' instead");
+    }
+  }
+
+  protected abstract void handleInfo(Attributes attributes) throws GameParseException;
+
+  protected abstract void handleLoader(Attributes attributes) throws GameParseException;
```

Comment:
I think we can gain a bit by applying open-close here, and could remove the inheritance heirarchy that these abstract methods would require. Noting that there is no `TripleaHandlerTest.java`, there is evidence that this was difficult to test and open-close should help fix that here.

Noting we have a common param of `Attributes`, and above we are mapping strings to these methods, it seems then that we could create a class like:
```
public interface XmlAttributeParser {
  String getXmlNodeName(); // preferably this would return an enum, but as an example in context it would be a String.
  void handle(Attributes attributes) throws GameParseException;
}
```

We could then update the constructor to init a list of these:
```
public TripleAHandler(GameData gameData) {
  this(
     gameData,
     new DiceSidesParser(),
     new TerritoryParser(),
     new connectionParser(),
     new ResourceParser(), 
    :
    :
}
@VisibleForTesting
TripleAHandler(GameData gameData, XmlAttributeParser ... parsers) {
   this.gameData = gameData;
   this.parsers = Arrays.asList(parsers);
}
```

Then handleElements would be simpler and not need to update when we change parsers:
```
 private void handleElement(String qname, Attributes attributes) throws SAXException {
    parsers.stream().
        .filter(parser -> parser.getXmlNodeName.equals(qname))
        .findFirst(parser -> parser.handle(attributes)
        .orElseThrow(() -> new RuntimeException("could not find parser for: " + qname + ", parsers loaded: " + parsers));
 }
```

Finally, the test burden for each parser would be much less, we could test each of the parsers independently and would only need to test the framework/handler code once. This would let us more thoroughly test the error/edge cases of each component as well. With the framework and handlers coupled, each handler needs to be tested via every path that the framework can take, meaning any handler test cases are multiplied by each such path. Splitting those two components allows us to test each individually and their respective test cases would no longer be related.